core/src/main/scala/org/ensime/core/Indexer.scala

```diff
@@ -24,10 +24,26 @@ class Indexer(
     LineSourcePositionHelper.fromFqnSymbol(hit)(vfs)
   )
 
-  def oldSearchTypes(query: String, max: Int) =
-    index.searchClasses(query, max).filterNot {
-      name => name.fqn.endsWith("$") || name.fqn.endsWith("$class")
-    }.map(typeResult)
+  def oldSearchTypes(query: String, max: Int) = {
+    val scalaNames = scala.collection.mutable.Set.empty[String]
+
+    def isUnique(fqn: String): Boolean = {
+      val stripped =
+        if (fqn.endsWith("$")) fqn.dropRight(1)
```

Comment:
Just experience. replaceAll is java method where's drop etc are scala EnrichedString methods that treat String as Seq[Char]